src/core/udb-api.service.js

```diff
@@ -230,6 +231,40 @@ function UdbApi(
     return deferredOffer.promise;
   };
 
+  /**
+   * @param {Array} events
+   * @return {Array}
+   */
+  this.reformatEvents = function(events) {
```

Comment:
This function looks rather complex. Is it covered by a unit test somewhere?

Reply:
Nope but  I could add one.
But better would be to have to have a consistent response when you do an API-call. 

Now when you do an api call **without** the param embed=true, you receive a response with an '**@type**' property.
<img width="1132" alt="Screenshot 2019-08-07 at 11 09 38" src="https://user-images.githubusercontent.com/9402377/62610553-331c2100-b904-11e9-8768-17a6a6687c15.png">



However when you do an api call with the param embed=true, you receive a response with an '**@context**' property instead of '@type'.
<img width="1136" alt="Screenshot 2019-08-07 at 11 10 04" src="https://user-images.githubusercontent.com/9402377/62610589-43340080-b904-11e9-8008-758f33b62875.png">

Reply:
Technically both are actually wrong, as the response should ideally include both `@type` and `@context`. `@context` is less valuable for our case though. It's mostly supposed to be used by crawlers like Google to look up the JSON-LD schema of the entity.

In the `embed=false` response it is relatively easy to include both. The `embed=true` response is a bit harder because right now it just includes the JSON-LD as you would get it from the detail endpoint (`GET /event/{id}` on udb3) - which is a read model that's cached in Redis and not created on the fly. We could add some post-processing to it on SAPI3 though that adds the `@type` based on the `@context`, like you do here on the front-end instead.

Reply:
@brampauwelyn what I would propose based on the info above is to leave this workaround in the front-end for now, and add a ticket to the backlog to fix this inconsistency on UDB3 / SAPI3. Could you do this with the screenshots from your comment? @Stiksels will probably want to fix this in scope of the API cleanup.

Reply:
Also adding a unit test would be useful yes so we have some kind of documentation on the expected `@type` for a given `@context`. Then the logic is easy to move to the back-end later on.